torch/distributed/device_mesh.py

```diff
@@ -173,7 +173,7 @@ class DeviceMesh:
         """
 
         _device_type: str
-        _mesh: torch.Tensor
+        _global_rank_permutation: torch.Tensor
```

Comment:
As I mentioned, I welcome suggestions for better names.

Reply:
I personally don't like the idea of permutation. Shall we call it `_global_ranks_alloc`?